src/main/java/Prune/Parser.java

```diff
@@ -0,0 +1,255 @@
+package Prune;
+
+import Prune.Tasks.TaskList;
+import Prune.Exceptions.InvalidCommand;
+import Prune.Exceptions.InvalidTaskNumber;
+import Prune.Tasks.Deadline;
+import Prune.Tasks.Event;
+import Prune.Tasks.Task;
+import Prune.Tasks.ToDo;
+
+import java.util.Arrays;
+
+/**
+ * The Parser class processes user input and translates user's inputs into actions.
+ * It interacts with the TaskList and Storage classes to manage and store them.
+ */
+public class Parser {
+
+    public static final String DISPLAY_TASKS_COMMAND = "list";
+    public static final String MARK_DONE_COMMAND = "mark";
+    public static final String MARK_NOT_DONE_COMMAND = "unmark";
+    public static final String TODO_COMMAND = "todo";
+    public static final String DEADLINE_COMMAND = "deadline";
+    public static final String EVENT_COMMAND = "event";
+    public static final String DELETE_COMMAND = "delete";
+    public static final String FIND_COMMAND = "find";
+    public TaskList tasks;
+    public Storage storage;
+
+    /**
+     * Constructs a Parser object with a given TaskList and Storage.
+     *
+     * @param tasks   TaskList containing the tasks.
+     * @param storage Storage used to save the tasks.
+     */
+    public Parser(TaskList tasks, Storage storage) {
+        this.tasks = tasks;
+        this.storage = storage;
+    }
+
+    /**
+     * Processes the user input command and executes the corresponding task action.
+     * Handles user input commands.
+     *
+     * @param input Input string containing the command and task arguments.
+     * @throws InvalidCommand    If the command is unknown.
+     * @throws InvalidTaskNumber If the task number provided is invalid.
+     */
+    public void processInput(String input) throws InvalidCommand, InvalidTaskNumber {
+        String[] inputArray = input.split(" ", 2);
+        String command = inputArray[0];
+        Task task = null;
+
+        switch (command) {
+        case DISPLAY_TASKS_COMMAND:
+            this.displayTasks();
+            return;
+
+        case MARK_DONE_COMMAND:
+        case MARK_NOT_DONE_COMMAND:
+            task = processMarking(inputArray);
+            printMarkTaskMsg(task);
+            break;
+
+        case TODO_COMMAND:
+        case DEADLINE_COMMAND:
+        case EVENT_COMMAND:
+            task = processAdding(inputArray);
```

Comment:
better name would be `processAddCommand` rather than `processAdding`. 
similarly other methods.